src/rebar_git_resource.erl

```diff
@@ -306,11 +306,11 @@ collect_default_refcount(Dir) ->
                 case Tag of
                     undefined ->
                         AbortMsg2 = "Getting rev-list of git dependency failed in " ++ Dir,
-                        {ok, PatchLines} = rebar_utils:sh("git rev-list HEAD",
+                        {ok, PatchLines} = rebar_utils:sh("git rev-list --count HEAD",
                                                           [{use_stdout, false},
                                                            {cd, Dir},
                                                            {debug_abort_on_error, AbortMsg2}]),
-                        rebar_utils:line_count(PatchLines);
+                        {ok, list_to_integer(string:trim(PatchLines))};
```

Comment:
Ah didn't notice this last time around. `string:trim/1` is only being used starting in OTP-20. We're a few months from dropping OTP-19 compatibility. You might want to use `rebar_string:trim/1` instead, which maintains compatibility and makes this mergeable today.

Reply:
Shouldn't the OTP 19 build check catch this?

Reply:
(Just curious, fix is on its way.)

Reply:
Unfortunately, we don't really have tests for the git fetching providers because they'd require depending on external repositories. Most of that stuff is mocked out so we have to just be doubly careful about it. We also have integration tests at https://github.com/tsloughter/rebar3_tests that do actual entire fetches.

Reply:
FWIW it appears you patched only one of the two calls.

Reply:
:facepalm: